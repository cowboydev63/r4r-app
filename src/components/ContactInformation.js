import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';

class ContactInformation extends React.PureComponent {
    static propTypes = {
        contact: PropTypes.shape({
            name: PropTypes.shape({
                prefix: PropTypes.string,
                firstName: PropTypes.string,
                middleName: PropTypes.string,
                lastName: PropTypes.string,
                suffix: PropTypes.string,
            }),
            title: PropTypes.string,
            phone: PropTypes.string,
            email: PropTypes.string,
        }),
    }

    render() {
        if(!this.props.contact || !this.props.contact.name) {
            return null;
        }
        const {
            name: {
                prefix,
                firstName,
                middleName,
                lastName,
                suffix
            },
            title,
            phone,
            email
        } = this.props.contact;
        return (
            <Theme element="address" className='contact-information'>
                { firstName || lastName ? <p>{[prefix, firstName, middleName, lastName, suffix].join(' ')}</p> : null }
                { title ? <p>{ title }</p> : null }
                { phone ? <p>{ phone }</p> : null }
                { email ? <a href={`mailto:${ email }`}>{ email }</a> : null }
            </Theme>
        )
    }
}

export default ContactInformation;