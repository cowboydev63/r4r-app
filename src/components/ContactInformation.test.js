import React from 'react';
import { shallow } from 'enzyme';
import ContactInformation from './ContactInformation';

describe('ContactInfomation Component', () => {
  it('renders without crashing', () => {
    shallow(<ContactInformation />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<ContactInformation />);
    expect(wrapper).toMatchSnapshot();
  })
})