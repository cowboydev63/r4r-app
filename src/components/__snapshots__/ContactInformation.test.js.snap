// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfomation Component renders correctly 1`] = `""`;
